mpas_analysis/shared/climatology/climatology.py

```diff
@@ -862,7 +863,10 @@ def _setup_climatology_caching(ds, startYearClimo, endYearClimo,
 
         cacheIndex = len(cacheInfo)
         for year in years:
-            cacheIndices[yearsInDs == year] = cacheIndex
+            for month in monthValues:
+                mask = numpy.logical_and(yearsInDs == year,
+                                         monthsInDs == month)
+                cacheIndices[mask] = cacheIndex
```

Comment:
`cacheIndices` is an array that tells you which cache file each time index corresponds with.  With the change I made, the cache index is left at -1 if a month is not in monthsValues (so it won't be in any cache file).